WireDomain/Sources/WireDomain/Notifications/Builders/NewSystemMessageNotificationBuilder.swift

```diff
@@ -0,0 +1,194 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireAPI
+import WireDataModel
+
+struct NewSystemMessageNotificationBuilder: NotificationBuilder {
+
+    enum SystemMessage {
+        case memberLeave(removedUserIDs: Set<UUID>)
+        case conversationCreated
+        case memberJoin(addedUserIDs: Set<UUID>)
+        case conversationDeleted
+        case messageTimerUpdate(newTimer: Int64?)
+    }
+
+    private let context: Context
+
+    struct Context {
+        let senderName: String?
+        let conversationName: String?
+        let isGroupConversation: Bool
+        let teamName: String?
+        let conversationID: WireAPI.QualifiedID
+        let senderID: UUID
+        let selfUserID: UUID
+        let hidesNotificationContent: Bool
+        let systemMessage: SystemMessage
+    }
+
+    init(
+        systemMessage: SystemMessage,
+        conversationID: WireAPI.QualifiedID,
+        senderID: UserID
+    ) async {
```

Comment:
I find it a bit weird that the init needs to be async and already does many things.
What if you inject the builder somewhere but it is never used?

I usually use the init to just prepare everything for the actual build-invocations.